ipn/ipnlocal/cert_test.go

```diff
@@ -26,3 +33,64 @@ func TestValidLookingCertDomain(t *testing.T) {
 		}
 	}
 }
+
+//go:embed testdata/*
+var certTestFS embed.FS
+
+func TestCertStoreRoundTrip(t *testing.T) {
+	const testDomain = "example.com"
+
+	// Use a fixed verification timestamp so validity doesn't fall off when the
+	// cert expires. If you update the test data below, this may also need to be
+	// updated.
+	testNow := time.Date(2023, time.February, 10, 0, 0, 0, 0, time.UTC)
+
+	// To re-generate a root certificate and domain certificate for testing,
+	// use:
+	//
+	//   	go run filippo.io/mkcert@latest example.com
+	//
+	// The content is not important except to be structurally valid so we can be
+	// sure the round-trip succeeds.
+	testRoot, err := certTestFS.ReadFile("testdata/rootCA.pem")
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
so you can do
```
roots := x509.NewCertPool
if !roots.AppendCertsFromPEM(extraRoots) {
  t.Fatal("cert check: failed append certificates")
  return false
}
```

Reply:
Almost—but we have to leave `roots == nil` for the production case, because otherwise we won't get the system cert pool (which is the default).

Reply:
(see above, I misunderstood what you were asking for here, but I get it now 🙂)